unlock-app/src/pages/_app.js

```diff
@@ -14,13 +14,15 @@ import web3Middleware from '../middlewares/web3Middleware'
 import currencyConversionMiddleware from '../middlewares/currencyConversionMiddleware'
 import storageMiddleware from '../middlewares/storageMiddleware'
 import walletMiddleware from '../middlewares/walletMiddleware'
+import postMessageMiddleware from '../middlewares/postMessageMiddleware'
 
 const config = configure()
 
 const __NEXT_REDUX_STORE__ = '__NEXT_REDUX_STORE__'
 
 function getOrCreateStore(initialState, history) {
   const middlewares = [
+    postMessageMiddleware(global),
```

Comment:
I think we need to have a discussion about the order of middlewares because it matters, right?
Also why pass `global` here?

Reply:
This allows the middleware to only use local variables, making it pure. If we access window directly, it makes it much harder to test, much easier to screw up and so on.

Reply:
I forgot to answer your first question. The order matters if/only if middlewares are swallowing actions (not returning `next(action)`. Since this one may cause a url change, I figured having it first will improve performance in that case (user clicks or taps, and the page changes immediately), but it still passes the actions on transparently. This was very intentional so that the order this is inserted in the chain will not matter to the other middlewares

Reply:
Thanks!